scholarly/scholarly.py

```diff
@@ -108,54 +108,57 @@ def use_proxy(http='socks5://127.0.0.1:9050', https='socks5://127.0.0.1:9050'):
 
 def use_tor():
     logger.info("Setting tor as the proxy")
-    use_proxy(http='socks5://127.0.0.1:9050', https='socks5://127.0.0.1:9050')
+    use_proxy(http=_TOR_SOCK,
+              https=_TOR_SOCK)
 
 
 def _has_captcha(text):
     flags = ["Please show you're not a robot",
              "network may be sending automated queries",
              "have detected unusual traffic from your computer",
              "scholarly_captcha"]
-
-    if any([i in text for i in flags]):
-        return True
-    return False
+    return any([i in text for i in flags])
 
 
 def _get_page(pagerequest):
     """Return the data for a page on scholar.google.com"""
     logger.info("Getting %s", pagerequest)
     # Delay for avoiding overloading scholar
     time.sleep(1+random.uniform(0, 1))
+    resp = None
+    while True:
+        # If Tor is running we use the proxy
+        with requests.Session() as session:
+            if _TOR_WORKS:
+                # Tor uses the 9050 port as the default socks port
+                session.proxies = {'http':  _TOR_SOCK,
+                                   'https': _TOR_SOCK}
+            else:
+                session.proxies = _PROXIES
 
-    # If Tor is running we use the proxy
-    with requests.Session() as session:
-        if _TOR_WORKS:
-            # Tor uses the 9050 port as the default socks port
-            session.proxies = {'http':  'socks5://127.0.0.1:9050',
-                               'https': 'socks5://127.0.0.1:9050'}
-        else:
-            session.proxies = _PROXIES
-
-        try:
-            resp = session.get(pagerequest, headers=_HEADERS, cookies=_COOKIES, timeout=_TIMEOUT)
-            if resp.status_code == 200:
-                if _has_captcha(resp.text):
-                    logger.info("Got a CAPTCHA. Retrying...")
+            try:
+                _HEADERS['User-Agent'] = UserAgent().random
+                resp = session.get(pagerequest,
+                                   headers=_HEADERS,
+                                   cookies=_COOKIES,
+                                   timeout=_TIMEOUT)
+                if resp.status_code == 200:
+                    if _has_captcha(resp.text):
+                        logger.info("Got a CAPTCHA. Retrying...")
+                    else:
+                        break
```

Comment:
This is risky. We get out of the loop with an incorrect response, but we are not going to refresh the Tor ID as we need an exception for the refresh to be triggered

Reply:
Perhaps raising an exception would be better.

Reply:
I have added an exception raise for consistency.
Also added *.log to gitignore.

Reply:
Need to raise an Exception also at 
```
logger.info(f"""Got a response code {resp.status_code}. Retrying...""")
```